packages/fether-electron/src/main/app/methods/setupRequestListeners.js

```diff
@@ -30,6 +40,59 @@ function setupRequestListeners (fetherApp) {
       callback({ requestHeaders: details.requestHeaders }); // eslint-disable-line
     }
   );
+
+  // Content Security Policy (CSP)
+  session.defaultSession.webRequest.onHeadersReceived((details, callback) => {
+    /* eslint-disable */
+    callback({
+      responseHeaders: {
+        ...details.responseHeaders,
+        "Content-Security-Policy": [CSP]
+      }
+    });
+    /* eslint-enable */
+  });
+
+  /**
+   * Limit specific permissions (i.e. `openExternal`) in response to events from particular origins
+   * to limit the exploitability of applications that load remote content.
+   *
+   * References:
+   * https://electronjs.org/docs/api/session#sessetpermissionrequesthandlerhandler
+   * https://doyensec.com/resources/us-17-Carettoni-Electronegativity-A-Study-Of-Electron-Security-wp.pdf
+   */
+  session.defaultSession.setPermissionRequestHandler(
```

Comment:
When is this function called? I tried putting a console.log inside, and I never saw the log, even after clicking on links inside fether

Reply:
See response here https://github.com/paritytech/fether/pull/451#issuecomment-476249550